src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -1044,4 +1047,42 @@ public void replaceDownloadedFile(String linkToDownloadedFile, LinkedFile downlo
 
         this.setFiles(linkedFiles);
     }
+
+    /**
+     * Merge this entry's fields with another BibEntry. Non-intersecting fields will be automatically merged. In cases of
+     * intersection, priority is given to THIS entry's field value.
+     *
+     * @param other another BibEntry from which fields are sourced from
+     */
+    public void mergeWith(BibEntry other) {
+        mergeWith(other, Set.of());
+    }
+
+    /**
+     * Merge this entry's fields with another BibEntry. Non-intersecting fields will be automatically merged. In cases of
+     * intersection, priority is given to THIS entry's field value, UNLESS specified otherwise in the arguments.
+     *
+     * @param other another BibEntry from which fields are sourced from
+     * @param otherPrioritizedFields collection of Fields in which 'other' has a priority into final result
+     */
+    public void mergeWith(BibEntry other, Set<Field> otherPrioritizedFields) {
+        Set<Field> thisFields = new TreeSet<>(Comparator.comparing(Field::getName));
+        Set<Field> otherFields = new TreeSet<>(Comparator.comparing(Field::getName));
+
+        thisFields.addAll(this.getFields());
+        otherFields.addAll(other.getFields());
+
+        Set<String> thisFieldsNames = thisFields.stream().map(Field::getName).collect(Collectors.toSet());
+        Set<String> otherPrioritizedFieldsNames = otherPrioritizedFields.stream().map(Field::getName).collect(Collectors.toSet());
+
+        for (Field otherField : otherFields) {
+            // As iterator only goes through non-null fields from OTHER, the value below can never be null
+            String otherFieldValue = other.getField(otherField).orElse(null);
```

Comment:
Why not simply `.get()` instead of `.orElse(null)`?

Reply:
Ok, changed to only `.getField()`. I was trying to avoid a warning for calling a `get()` on an optional.